js/showAllSongs.js

```diff
@@ -20,40 +21,91 @@ var comparator = function(property) {
   };
 };
 
-window.onload = function() {
+const sortSongsByArtist = function(songs) {
   const allSongs = {};
+  songs.map(function(song) {
+    // Sorting by artist
+    const artist = song["artist"];
+
+    if (allSongs.hasOwnProperty(artist)) {
+      allSongs[artist].push(song);
+    } else {
+      allSongs[artist] = [song];
+    }
+  });
+
+  let allSongsByArtist = [];
+  for (let artist in allSongs) {
+    // Song titles in alphabetical order
+    const songsByArtist = allSongs[artist].sort(comparator("title"));
+    allSongsByArtist.push({ "artist": artist, "songs": songsByArtist });
+  }
+  // Artists in alphabetical order
+  allSongsByArtist = allSongsByArtist.sort(comparator("artist"));
+  return allSongsByArtist;
+};
+
+const ALL_TAG = "all";
+let selectedTag = ALL_TAG;
+const allSongsByTag = {};
```

Comment:
Generally speaking, anything with a dynamic set of keys, where you would use lots of computed property accesses and `hasOwnProperty`, probably wanted to be a `Map` instead of an `Object`.